addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -166,7 +174,34 @@ public void run() {
                     logger.error("An exception occurred while executing '{}' : '{}'",
                             new Object[] { commandLine.toString(), e.getMessage() });
                     updateState(RUN, OnOffType.OFF);
-                    updateState(OUTPUT, new StringType(e.getMessage()));
+
+                    if (e.getMessage().equals("OPEN")) {
+                        updateState(CONTACT_OUTPUT, OpenClosedType.OPEN);
+                    }
+
+                    if (e.getMessage().equals("CLOSED")) {
+                        updateState(CONTACT_OUTPUT, OpenClosedType.CLOSED);
+                    }
+
+                    if (e.getMessage().equals("ON")) {
+                        updateState(DIMMER_OUTPUT, OnOffType.ON);
+                        updateState(SWITCH_OUTPUT, OnOffType.ON);
+                    }
+
+                    if (e.getMessage().equals("OFF")) {
+                        updateState(DIMMER_OUTPUT, OnOffType.OFF);
+                        updateState(SWITCH_OUTPUT, OnOffType.OFF);
+                    }
+
+                    if (e.getMessage().equals("UP")) {
+                        updateState(ROLLERSHUTTER_OUTPUT, UpDownType.UP);
+                    }
+
+                    if (e.getMessage().equals("DOWN")) {
+                        updateState(ROLLERSHUTTER_OUTPUT, UpDownType.DOWN);
+                    }
+
+                    updateState(STRING_OUTPUT, new StringType(e.getMessage()));
```

Comment:
When the external scripts returns a value between 0 and 100, the output should be transformed to the OH2 datatype Percent. Then the exec2-binding could be used in combination with Rollershutters as it was the case for the old exec binding.